src/basicengine.cpp

```diff
@@ -189,38 +189,48 @@ int BasicEngine::executeTask(GreasyTask *task, int worker) {
     node = masterHostname;
   } else {
     node = workerNodes[worker];
+  }
```

Comment:
This part changes the logic of the `srun` command call, as explained in the description.

Reply:
We never faced issues here but the change makes sense.